src/internal/connector/onedrive/collections.go

```diff
@@ -376,8 +376,26 @@ func (c *Collections) UpdateCollections(
 	excluded map[string]struct{},
 	invalidPrevDelta bool,
 ) error {
+	// This can only be made local to a page as the folder rename
+	// might happen between pages. Every page in the graph response
+	// will have all the parents for that item.
+	visitedPaths := map[string]string{}
+
 	for _, item := range items {
 		if item.GetRoot() != nil {
+			collectionPath, err := GetCanonicalPath(
+				// TODO(meain): Is there a better way to get root path?
+				fmt.Sprintf("/drives/%s/root:", driveID),
```

Comment:
unfortunately I don't think so. Though maybe we should pull the pattern itself into a const and just reference it here?
```suggestion
				// TODO(meain): Is there a better way to get root path?
				fmt.Sprintf(rootDrivePattern, driveID),
```

Reply:
hmmm also the comments in the code may be off on what this path actually is. Looking at returned results in Graph Explorer I see stuff like the below which seems to imply we should use "drive" instead of "drives"
```
"parentReference": {
    "driveType": "business",
    "driveId": "some-drive-ID",
    "id": "an-ID",
    "path": "/drive/root:"
},
```

Reply:
Which query is this for? I am getting `drives` in the response. In fact I had changed some references of `drive` to  `drives` in the test assuming they were misspelt. This is from a delta response:
``` json
"parentReference": {
    "driveType": "business",
    "driveId": "b!54S....8PlFxA",
    "id": "01GI....LRRZ",
    "path": "/drives/b!54S....8PlFxA/root:"
},
```

Reply:
oh dear that's confusing :/ The query I pulled my data from was `https://graph.microsoft.com/v1.0/me/drive/root/delta` in Graph Explorer and signed into the dev account assigned to me

Looking at the code and using the query in there `https://graph.microsoft.com/v1.0/drives/<drive ID>/root/delta` yields paths with `drives` instead of `drive` like you've been seeing. I guess I should be more careful with the queries that I use 😅